awx/main/tasks.py

```diff
@@ -2707,6 +2703,130 @@ def pre_run_hook(self, inventory_update, private_data_dir):
+            options['exclude_empty_groups'] = True
+        if getattr(settings, '%s_INSTANCE_ID_VAR' % src.upper(), False):
+            options['instance_id_var'] = getattr(settings, '%s_INSTANCE_ID_VAR' % src.upper())
+
+        # Verbosity is applied to saving process, as well as ansible-inventory CLI option
+        if inventory_update.verbosity:
+            options['verbosity'] = inventory_update.verbosity
+
+        # Mock ansible-runner events
+        class CallbackHandler(logging.Handler):
+            def __init__(self, event_handler, cancel_callback, job_timeout, verbosity,
+                         counter=0, **kwargs):
+                self.event_handler = event_handler
+                self.cancel_callback = cancel_callback
+                self.job_timeout = job_timeout
+                self.job_start = time.time()
+                self.last_check = self.job_start
+                # TODO: we do not have events from the ansible-inventory process
+                # so there is no way to know initial counter of start line
+                self.counter = counter
+                self.skip_level = [logging.WARNING, logging.INFO, logging.DEBUG, 0][verbosity]
+                self._start_line = 0
+                super(CallbackHandler, self).__init__(**kwargs)
+
+            def emit(self, record):
+                this_time = time.time()
+                if this_time - self.last_check > 0.5:
+                    self.last_check = this_time
+                    if self.cancel_callback():
+                        raise RuntimeError('Inventory update has been canceled')
+                if self.job_timeout and ((this_time - self.job_start) > self.job_timeout):
+                    raise RuntimeError('Inventory update has timed out')
+
+                # skip logging for low severity logs
+                if record.levelno < self.skip_level:
+                    return
+
+                self.counter += 1
+                msg = self.format(record)
+                n_lines = msg.strip().count('\n')  # don't count new-lines at boundry of text
+                dispatch_data = dict(
+                    created=now().isoformat(),
+                    event='verbose',
+                    counter=self.counter,
+                    stdout=msg + '\n',
+                    start_line=self._start_line,
+                    end_line=self._start_line + n_lines
+                )
+                self._start_line += n_lines + 1
+
+                self.event_handler(dispatch_data)
+
+        handler = CallbackHandler(
+            self.event_handler, self.cancel_callback,
+            verbosity=inventory_update.verbosity,
+            job_timeout=self.get_instance_timeout(self.instance),
+            counter=self.event_ct
+        )
+        inv_logger = logging.getLogger('awx.main.commands.inventory_import')
+        handler.formatter = inv_logger.handlers[0].formatter
+        inv_logger.handlers[0] = handler
```

Comment:
After this is finished, we should probably reset this logger back to how it was before.